Userland/Libraries/LibJS/Runtime/Iterator.cpp

```diff
@@ -96,29 +108,24 @@ ThrowCompletionOr<NonnullGCPtr<IteratorRecord>> get_iterator(VM& vm, Value objec
     return TRY(get_iterator_from_method(vm, object, *method));
 }
 
-// 2.1.1 GetIteratorDirect ( obj ), https://tc39.es/proposal-iterator-helpers/#sec-getiteratorflattenable
-ThrowCompletionOr<NonnullGCPtr<IteratorRecord>> get_iterator_direct(VM& vm, Object& object)
-{
-    // 1. Let nextMethod be ? Get(obj, "next").
-    auto next_method = TRY(object.get(vm.names.next));
-
-    // 2. Let iteratorRecord be Record { [[Iterator]]: obj, [[NextMethod]]: nextMethod, [[Done]]: false }.
-    // 3. Return iteratorRecord.
-    auto& realm = *vm.current_realm();
-    return vm.heap().allocate<IteratorRecord>(realm, realm, object, next_method, false);
-}
-
-// 2.1.2 GetIteratorFlattenable ( obj, stringHandling ), https://tc39.es/proposal-iterator-helpers/#sec-getiteratorflattenable
-ThrowCompletionOr<NonnullGCPtr<IteratorRecord>> get_iterator_flattenable(VM& vm, Value object, StringHandling string_handling)
+// 7.4.5 GetIteratorFlattenable ( obj, primitiveHandling ), https://tc39.es/ecma262/#sec-getiteratorflattenable
+ThrowCompletionOr<NonnullGCPtr<IteratorRecord>> get_iterator_flattenable(VM& vm, Value object, PrimitiveHandling primitive_handling)
 {
     // 1. If obj is not an Object, then
     if (!object.is_object()) {
-        // a. If stringHandling is reject-strings or obj is not a String, throw a TypeError exception.
-        if (string_handling == StringHandling::RejectStrings || !object.is_string())
+        // a. If primitiveHandling is reject-primitives, throw a TypeError exception.
+        if (primitive_handling == PrimitiveHandling::RejectPrimitives)
             return vm.throw_completion<TypeError>(ErrorType::NotAnObject, object.to_string_without_side_effects());
+
+        // b. Assert: primitiveHandling is iterate-string-primitives.
+        ASSERT(primitive_handling == PrimitiveHandling::IterateStringPrimitives);
```

Comment:
Did you intentionally use `ASSERT` here? As far as I understand, most places use `VERIFY` and `ASSERT` is for perf-sensitive code (?)